api/cpp/include/slint_sharedvector.h

```diff
@@ -21,17 +22,49 @@ struct SharedVector
                 cbindgen_private::slint_shared_vector_empty())))
     {
     }
+
     /// Creates a new vector that holds all the elements of the given std::initializer_list \a args.
-    SharedVector(std::initializer_list<T> args) : SharedVector()
+    SharedVector(std::initializer_list<T> args)
+        : SharedVector(SharedVector::with_capacity(args.size()))
     {
-        auto new_array = SharedVector::with_capacity(args.size());
-        auto new_data = reinterpret_cast<T *>(new_array.inner + 1);
+        auto new_data = reinterpret_cast<T *>(inner + 1);
         auto input_it = args.begin();
         for (std::size_t i = 0; i < args.size(); ++i, ++input_it) {
             new (new_data + i) T(*input_it);
-            new_array.inner->size++;
+            inner->size++;
+        }
+    }
+
+    /// Creates a vector of a given size, with default-constructed data.
+    SharedVector(size_t size) : SharedVector(SharedVector::with_capacity(size))
```

Comment:
I suggest to make this constructor explicit. (the one from std::vector is, too, and I can't imagine this being a useful implicit conversion)

Reply:
right, what was i thinking?